### PR TITLE
Add documentation for configuration options

### PR DESCRIPTION
**Pull request by derekprior**
Added Yard configuration and documentation for
`Clearance::Configuration`. Moving forward, we will be adding
documentation to the rest of Clearance.

### CODE REVIEW

File: lib/clearance/configuration.rb [hunk 1]
```diff
@@ -44,23 +121,36 @@ def  user_actions
...
   end
 
+  # Set Clearance's configuration
+  # @param config [Clearance::Configuration]
   def self.configuration=(config)
     @configuration = config
   end
 
+  # Modify Clearance's current configuration
+  #   Clearance.configure do |config|
```
Comment (mcmire) [inline lib/clearance/configuration.rb]: This code sample might not render correctly, you might have to indent this by four spaces to do so.

File: lib/clearance/configuration.rb [hunk 2]
```diff
@@ -1,23 +1,94 @@
...
+    # live environments to prevent session hijacking. For more, see
+    # [RFC6265](http://tools.ietf.org/html/rfc6265#section-5.2.5).
+    # @return [Boolean]
+    attr_accessor :secure_cookie
+
+    # The array of sign in guards to run when signing a user in.
+    # Defaults to an empty array. Sign in guards respond to `call` and are
+    # initialized with a session and the current stack. Each guard can decide
+    # to fail the sign in, yield to the next guard, or allow the sign in.
+    # @return [Array #call]
```
Comment (mcmire) [inline lib/clearance/configuration.rb]: What does `Array #call` mean here? (Just curious)

File: .yardopts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+--exclude templates
+--markup markdown
+--markup-provider redcarpet
```
Comment (mcmire) [inline .yardopts]: You might also want to include:

```
--no-private
--protected
```

File: lib/clearance/configuration.rb (hunk 2 above)
Comment (derekprior) [inline lib/clearance/configuration.rb]: I'm trying to say it's an array of things that respond to `#call`. But it doesn't actually seem to work the way I'm lead to believe it works here: http://yardoc.org/types.html

**Comment by mcmire** [comment]
I am not a YARD expert, I just use a few of the features, but other than the comment above, everything seems to look good here.

File: lib/clearance/configuration.rb (hunk 2 above)
Comment (gabebw) [inline lib/clearance/configuration.rb]: Maybe you want `Array<#call>`?

**Pull request merged**